src/systems/sensors/Sensors.cc

```diff
@@ -473,7 +473,26 @@ std::string Sensors::CreateSensor(const Entity &_entity,
   }
 
   // Create within ign-sensors
-  auto sensorId = this->dataPtr->sensorManager.CreateSensor(_sdf);
+  ignition::sensors::SensorId sensorId;
+  if (_sdf.Type() == sdf::SensorType::CAMERA)
+  {
+    std::unique_ptr<sensors::CameraSensor> cameraSensor =
+      std::make_unique<sensors::CameraSensor>();
+    cameraSensor->Load(_sdf);
+    sensorId = this->dataPtr->sensorManager.AddSensor(std::move(cameraSensor));
+  }
+  else if (_sdf.Type() == sdf::SensorType::THERMAL_CAMERA)
```

Comment:
depth and rgbd camera classes are not being used, we don't need to link against them, we can use `createSensor`. If need to modify some of the specific parameters, then we need to instanciate a class and create a specific case for them.

Reply:
you can try `sensors_demos.sdf` which is creating: `camera`, `depth_camera`, `gpu_lidar`, `rgbd_camera` and `thermal_camera`

Reply:
Since this is now targeting Edifice, I think lines 485 - 504 can be replaced with: 
```
ignition::sensors::SensorId sensorId = this->dataPtr->sensorManager.CreateSensor(_sdf);
```

At least, this worked for me.

Reply:
I took the liberty of applying this change in 2f674e6736a5e7a726187f01963c19bf943c6c00. Feel free to revert.